### PR TITLE
backport: merge bitcoin#22402, #23583, #23618, #17874, #21286, #23489, #23675, #24130, #23556, #23862, #24722, #24132, #24668, #25643, #25719, partial #22469, #23677 (qt backports)

### PR DESCRIPTION
**Pull request by kwvg**
## Additional Notes

* Depends on
  * https://github.com/dashpay/dash/pull/5355
  * https://github.com/dashpay/dash/pull/5408
  * https://github.com/dashpay/dash/pull/5426
  * https://github.com/dashpay/dash/pull/5448

### CODE REVIEW

**Comment by kwvg** [comment]
<details>
<summary>Guix Binaries:</summary>
<p>

```
92b52cb671e75ed8f5e6b7f60d8b83ed66d1c403d6df2cf09c9f29694dda54f7  guix-build-f1b09cf898a0/distsrc-f1b09cf898a0-aarch64-linux-gnu/src/dash-cli
ab311c4d186cc6195cfbd9c24f6e7dd2d8f2b0799760316910658a747110fad4  guix-build-f1b09cf898a0/distsrc-f1b09cf898a0-aarch64-linux-gnu/src/dashd
2fffa187736aa5a2223ac24076a8b60d5dc06b4e23094f11649305bcce0ce7cf  guix-build-f1b09cf898a0/distsrc-f1b09cf898a0-aarch64-linux-gnu/src/dash-tx
b484f0032de1a050ff57ca611fefb72d234651ba198927fa20afeb9653a434ef  guix-build-f1b09cf898a0/distsrc-f1b09cf898a0-aarch64-linux-gnu/src/dash-wallet
cdf7b6e4b6170ca590857ec45445b35b85fe0b73a7b3e754d274053241c3b188  guix-build-f1b09cf898a0/distsrc-f1b09cf898a0-aarch64-linux-gnu/src/qt/dash-qt
94912ade91e29a19db890869ac30b82725049556af1fe65cf6ba270645eb102c  guix-build-f1b09cf898a0/distsrc-f1b09cf898a0-aarch64-linux-gnu/src/test/test_dash
e4baa7afe1802788110881eaac274fc023abdd655567588e65da667d62248079  guix-build-f1b09cf898a0/distsrc-f1b09cf898a0-arm64-apple-darwin/src/dash-cli
945b5cc3ff79db2eb228b05344fbeac9be14a568f680e5f11d53a1ffea9622de  guix-build-f1b09cf898a0/distsrc-f1b09cf898a0-arm64-apple-darwin/src/dashd
9abf4e3f8fb0814687794622182caa8169eb540de9dcd502bec46ce9770bfe04  guix-build-f1b09cf898a0/distsrc-f1b09cf898a0-arm64-apple-darwin/src/dash-tx
f27ebb1b21bc296609e8e6cebc4b848b46ae4578b6bef2fd71997624a82324ec  guix-build-f1b09cf898a0/distsrc-f1b09cf898a0-arm64-apple-darwin/src/dash-wallet
ce01ee91b4814c25c1abd08fc230b7ee5c1cde4f37869589351db9921650b149  guix-build-f1b09cf898a0/distsrc-f1b09cf898a0-arm64-apple-darwin/src/qt/dash-qt
f25dffdc2a76774b5d05d0253b950c8797d8382f72b72a7a9f9db2e5db6527d9  guix-build-f1b09cf898a0/distsrc-f1b09cf898a0-arm64-apple-darwin/src/test/test_dash
73c964cc449b8469e1345f5c6a16a31cf13991737216c12fa030afee324a6c43  guix-build-f1b09cf898a0/distsrc-f1b09cf898a0-arm-linux-gnueabihf/src/dash-cli
f1901edbc2988ff1751211381e6a92a8695570471606a356a6c83ca14ffc5b01  guix-build-f1b09cf898a0/distsrc-f1b09cf898a0-arm-linux-gnueabihf/src/dashd
8983d644b8b9bab42e1241872a01243d15818c96506e9f187fe2f1ee64938565  guix-build-f1b09cf898a0/distsrc-f1b09cf898a0-arm-linux-gnueabihf/src/dash-tx
d4a64e1083f715050c96b964b0441a39643bff38c010381e4eec2a61a05f6d8a  guix-build-f1b09cf898a0/distsrc-f1b09cf898a0-arm-linux-gnueabihf/src/dash-wallet
728d3d5540a3e5eef3637bb1c27b522e0dca29ba2c9b9950006779b40346a72d  guix-build-f1b09cf898a0/distsrc-f1b09cf898a0-arm-linux-gnueabihf/src/qt/dash-qt
85fccb52899d85b0eb64513d63f517a7c605f3048953bd4194d80d82ef4769c2  guix-build-f1b09cf898a0/distsrc-f1b09cf898a0-arm-linux-gnueabihf/src/test/test_dash
17b79efad36e7b1bd8862009074ee27174d17c1807d2381690efc297cf457cb7  guix-build-f1b09cf898a0/distsrc-f1b09cf898a0-riscv64-linux-gnu/src/dash-cli
c06827b109ca1beec628d59dbe3b2a464a001e973923f4725df477335d70b8c5  guix-build-f1b09cf898a0/distsrc-f1b09cf898a0-riscv64-linux-gnu/src/dashd
f11c9ab01ae85a5ca6a7b504bfd1a04d28674b92c5142af55cbf1a1a21c5a912  guix-build-f1b09cf898a0/distsrc-f1b09cf898a0-riscv64-linux-gnu/src/dash-tx
34d70f9baa4c2236ca4a2c2ec0f31d6266ae7fb276e17a284bfbdf6548a1f9f2  guix-build-f1b09cf898a0/distsrc-f1b09cf898a0-riscv64-linux-gnu/src/dash-wallet
f151246d614b76063fa53fa3021c0b204840aaecbccf5cf10a31c4e90856bf04  guix-build-f1b09cf898a0/distsrc-f1b09cf898a0-riscv64-linux-gnu/src/qt/dash-qt
4550f5044af7e7ddd9075c72b0a1927225c76936ccbf76b758b6eb0e5d233b9f  guix-build-f1b09cf898a0/distsrc-f1b09cf898a0-riscv64-linux-gnu/src/test/test_dash
08c26c533125fee0c66e7f57581bcbeb40db18e6dbeda88265a9adaa4588ef3b  guix-build-f1b09cf898a0/distsrc-f1b09cf898a0-x86_64-apple-darwin/src/dash-cli
527e8cdc6543740134b4272ccc0848be5fb9c862430a54beb57cfe4ae6fc3244  guix-build-f1b09cf898a0/distsrc-f1b09cf898a0-x86_64-apple-darwin/src/dashd
dda40bddc9cc1b283e5b8ad58af99ad1b56936619a03c016c6633c392e6d282a  guix-build-f1b09cf898a0/distsrc-f1b09cf898a0-x86_64-apple-darwin/src/dash-tx
662021fa93620b8283069e28ff824e4dc0e7d59fc8bd7a08ac10b9649012b14c  guix-build-f1b09cf898a0/distsrc-f1b09cf898a0-x86_64-apple-darwin/src/dash-wallet
91ff24567076cc7bdff75040dc475d4f8b8eeefc6228eb5f2cf0f287482a8b3a  guix-build-f1b09cf898a0/distsrc-f1b09cf898a0-x86_64-apple-darwin/src/qt/dash-qt
4e2b1ff4e7c31215682110ace4a6b49a0209682dc73120200a95b099c24ca38d  guix-build-f1b09cf898a0/distsrc-f1b09cf898a0-x86_64-apple-darwin/src/test/test_dash
242e4fec6012244c3ea4ea97b0dca09c7a858eed66e5787bc1720eb811e5c703  guix-build-f1b09cf898a0/distsrc-f1b09cf898a0-x86_64-linux-gnu/src/dash-cli
59c50205ff7b8de37ceea4b468d5764268a5a8286e57d9c8bf69378732c95440  guix-build-f1b09cf898a0/distsrc-f1b09cf898a0-x86_64-linux-gnu/src/dashd
0a01546f4bca39f2fe1b1f998bc6ce377ae03d7860607b534107b30681cd0715  guix-build-f1b09cf898a0/distsrc-f1b09cf898a0-x86_64-linux-gnu/src/dash-tx
1b137ecee5d87c6f782e1259162bbc293a196fc66170a2602c9e6c78c24d4d32  guix-build-f1b09cf898a0/distsrc-f1b09cf898a0-x86_64-linux-gnu/src/dash-wallet
4af57b88243ce841a8c87313497516f88b2fac7e6096029cdd1f57006303c2ce  guix-build-f1b09cf898a0/distsrc-f1b09cf898a0-x86_64-linux-gnu/src/qt/dash-qt
499b28a76426f34b0359e5e9ca41b3eb60c89cd7fc93b94bb2805826e4144988  guix-build-f1b09cf898a0/distsrc-f1b09cf898a0-x86_64-linux-gnu/src/test/test_dash
e8f40e234f609af97c3b0734121927a7ea121de500b5961943936c81a4b35364  guix-build-f1b09cf898a0/distsrc-f1b09cf898a0-x86_64-w64-mingw32/src/dash-cli.exe
d94a4c43db838655e415a372d83fac6d761961bf737a3577e191c08ceea8774f  guix-build-f1b09cf898a0/distsrc-f1b09cf898a0-x86_64-w64-mingw32/src/dashd.exe
c81cfd12f07137e58b5edb3ac348b29301af9ec524e10c26b4bbe80b415619c7  guix-build-f1b09cf898a0/distsrc-f1b09cf898a0-x86_64-w64-mingw32/src/dash-tx.exe
041fb8b5843cf321f44dee76cc57ca250d7769ca56959b5fe191cb6a10bf91e6  guix-build-f1b09cf898a0/distsrc-f1b09cf898a0-x86_64-w64-mingw32/src/dash-wallet.exe
755b813480c653bc52c08702f398b68bc8dfa23b6b015c9a862f168fce510bfe  guix-build-f1b09cf898a0/distsrc-f1b09cf898a0-x86_64-w64-mingw32/src/qt/dash-qt.exe
7ec44ffc8d94fb362eaf6c525f11df625dfc3631ff2cab3450abca6875f2a30a  guix-build-f1b09cf898a0/distsrc-f1b09cf898a0-x86_64-w64-mingw32/src/test/test_dash.exe
```

</p></details>

<details>
<summary>Guix Archives:</summary>
<p>

```
0f61993d803e1985c7bdad86830529bdbe580606d5a759416c6456c9e0b9fc14  guix-build-f1b09cf898a0/output/aarch64-linux-gnu/dashcore-f1b09cf898a0-aarch64-linux-gnu-debug.tar.gz
bc37f8d57da6f8726977e24a8782ff260bd1fa8a3b1979017c8e72a7f8bbf463  guix-build-f1b09cf898a0/output/aarch64-linux-gnu/dashcore-f1b09cf898a0-aarch64-linux-gnu.tar.gz
d3ce61d8aa1ba73e3b400db86a5b419fe8fe0c15f716e4e090f78d0516a2027b  guix-build-f1b09cf898a0/output/arm64-apple-darwin/dashcore-f1b09cf898a0-arm64-apple-darwin.tar.gz
c36fe947b1b93928d8f032193f28a4ca5d1ba9d88b5b47bab1721db5e50de526  guix-build-f1b09cf898a0/output/arm64-apple-darwin/dashcore-f1b09cf898a0-arm64-apple-darwin-unsigned.dmg
19e57dfcdb3542b70393bd49df6f0758610f5cc3631d9362c101e871faf3ed22  guix-build-f1b09cf898a0/output/arm64-apple-darwin/dashcore-f1b09cf898a0-arm64-apple-darwin-unsigned.tar.gz
206324cd6b1a69f792822ee2e9a3e890b215568d374981357d2c93c5d9dd1496  guix-build-f1b09cf898a0/output/arm-linux-gnueabihf/dashcore-f1b09cf898a0-arm-linux-gnueabihf-debug.tar.gz
fafc01b91aa56bc502bf2f1245dbf26ae3bd73f53f53a8cf415a82fe9c72e85b  guix-build-f1b09cf898a0/output/arm-linux-gnueabihf/dashcore-f1b09cf898a0-arm-linux-gnueabihf.tar.gz
eac6ed0bc2cc4ad985fb7ab99a5006c4cf120e32b68dbac7dd3ead5ea9283219  guix-build-f1b09cf898a0/output/dist-archive/dashcore-f1b09cf898a0.tar.gz
cb41cdcb51a86073c1f20f068d1b832d7c59e310a0ef8d1aba3f362236894295  guix-build-f1b09cf898a0/output/riscv64-linux-gnu/dashcore-f1b09cf898a0-riscv64-linux-gnu-debug.tar.gz
9f86e5247d9fa4ec80f4e4a1c283cb0c67a15b22ba84a989dd86024e32c875cc  guix-build-f1b09cf898a0/output/riscv64-linux-gnu/dashcore-f1b09cf898a0-riscv64-linux-gnu.tar.gz
2f2598341d729291fd7d36365bb6330c3014c9f8277c0e694c6eb9d3cdc1b67a  guix-build-f1b09cf898a0/output/x86_64-apple-darwin/dashcore-f1b09cf898a0-x86_64-apple-darwin.tar.gz
a7b323716bb0dcc086c2acd3696e23f9ad61def5eadc993edb2c7443bb88e6f9  guix-build-f1b09cf898a0/output/x86_64-apple-darwin/dashcore-f1b09cf898a0-x86_64-apple-darwin-unsigned.dmg
1e07bc7a394094c7ac8021882eec65a685140197d81278ffa4eac2abb08af923  guix-build-f1b09cf898a0/output/x86_64-apple-darwin/dashcore-f1b09cf898a0-x86_64-apple-darwin-unsigned.tar.gz
1d2b454edf8e6643d9ef6664f3ddc0a042842daa8f3b0f4750ff55812edb637f  guix-build-f1b09cf898a0/output/x86_64-linux-gnu/dashcore-f1b09cf898a0-x86_64-linux-gnu-debug.tar.gz
58840b8b026c147363ab564c71bfdf172ec065a07850fd9f879fe5d45d9f2af1  guix-build-f1b09cf898a0/output/x86_64-linux-gnu/dashcore-f1b09cf898a0-x86_64-linux-gnu.tar.gz
728b15da891229b628d3444c53293fb07209e928387f4388dd4ea3bb06027a3f  guix-build-f1b09cf898a0/output/x86_64-w64-mingw32/dashcore-f1b09cf898a0-win64-debug.zip
ee9d99dea698fab637801c1199cfc92fea1d3e4af5f151f9fb424c92757ca667  guix-build-f1b09cf898a0/output/x86_64-w64-mingw32/dashcore-f1b09cf898a0-win64-setup-unsigned.exe
c6067b499faf8abc838114e4c6ecc2a4ad9471f7e5e97aab664f2c3195b0b51d  guix-build-f1b09cf898a0/output/x86_64-w64-mingw32/dashcore-f1b09cf898a0-win64-unsigned.tar.gz
7b57eb8b43ac39e4375e67b1c788cb2882b340dcba6f92bea95c735e2a796996  guix-build-f1b09cf898a0/output/x86_64-w64-mingw32/dashcore-f1b09cf898a0-win64.zip
```

</p></details>

File: depends/packages/qt.mk [hunk 1]
```diff
@@ -248,13 +248,13 @@ define $(package)_preprocess_cmds
   patch -p1 -i $($(package)_patch_dir)/fix_android_jni_static.patch && \
   patch -p1 -i $($(package)_patch_dir)/no-xlib.patch && \
   patch -p1 -i $($(package)_patch_dir)/dont_hardcode_x86_64.patch && \
-  patch -p1 -i $($(package)_patch_dir)/fix_limits_header.patch && \
```
Comment (knst) [inline depends/packages/qt.mk:240]: patch is removed, but file is still here

**Review by knst — CHANGES_REQUESTED** [review]
Could this PR be a good place for bitcoin#23675  to follow-up bitcoin#23489 changes? not necessary although.

----
note for review (no changes required): #23489 and #23556 are some how messed up together

It also explains why file `depends/patches/qt/fix_bigsur_style.patch` is missing in merge bitcoin#23489: Qt 5.15.2

File: depends/packages/qt.mk (hunk 1 above)
Comment (kwvg) [reply depends/packages/qt.mk:240]: Resolved in latest push

**Review by knst — APPROVED** [review]
utACK

**Review by UdjinM6 — APPROVED** [review]
Guix hashes match

Gitian
```
3ad4ab0f24d78f484be444f0089cd748c9eda06cd6bfd9d8f44239928af32206  dashcore-e573aea84206-aarch64-linux-gnu-debug.tar.gz
e89e3e84f4a3a552f662a02d98394816dd2e80ca846f1bac7a0ce32df17b52cf  dashcore-e573aea84206-aarch64-linux-gnu.tar.gz
90a72a350d5cb87a1349dda0a2b3576d9855b236bc563eeba52b4c8a52a84600  dashcore-e573aea84206-riscv64-linux-gnu-debug.tar.gz
e562aeb5e4c15eacab604f2cba5a84e60d0407b4d4fcae86da185d12b67c5182  dashcore-e573aea84206-riscv64-linux-gnu.tar.gz
06261b33b425e47240856941618984723167d76b5e6dc282c7b9fd470c83dfb0  dashcore-e573aea84206-x86_64-linux-gnu-debug.tar.gz
57d50eb4f49994b55b722e42a02e8bd2994e994653e8fec0b3fd13f08950e1e9  dashcore-e573aea84206-x86_64-linux-gnu.tar.gz
abe1e15bde3bdd152a3a5f57cd7270f4525a81c3bf91188dea5237114bce0652  dashcore-e573aea84206-win-unsigned.tar.gz
a7a614220f3bed587eadfa91a22a49a93587c7bea905fd85c783439f8c9be27d  dashcore-e573aea84206-win64-debug.zip
b5877cfce7f7f2a85010103fb2b6a0825d01e15fe36313e809a41dbc2051d2f2  dashcore-e573aea84206-win64-setup-unsigned.exe
5cf76b591a81cca88a0f403e5543626f689751b9efb5ee151382ce674493cc27  dashcore-e573aea84206-win64.zip
95ce15715216ed8eb40e1995149606d0d8f32a3eab83442265c3b3d4a04c6ed4  dashcore-e573aea84206-osx-unsigned.dmg
c1cb011fa888dfd2fb2d21479c4a38ade6729ed8951a7afb1ed1e95cf59c4232  dashcore-e573aea84206-osx-unsigned.tar.gz
8d28e451e9bfe0b7c11fc6e8b16057b23c9fe57619ab457fdcb8af729f27fc86  dashcore-e573aea84206-osx64-debug.tar.gz
c79728ec8a36cae526df81eff1945552dc072a31b14af25b33539f6c2a75ce52  dashcore-e573aea84206-osx64.tar.gz
```

utACK

**Review by PastaPastaPasta — APPROVED** [review]
utACK for merging via merge commit

**Pull request merged**